src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -347,6 +347,12 @@
         name="integer_literal",
         type="literal",
     ),
+    SystemFunctionName=RegexParser(
+        r"SYSTEM\$([A-z0-9]*)",
```

Comment:
This looks like a typo, but I don't think it is because it does match `_` as well.

Should this be more explicit:

```suggestion
        r"SYSTEM\$([A-Za-z0-9_]*)",
```

Reply:
Good note, implemented.